src/main/java/org/spongepowered/api/service/permission/PermissionDescription.java

```diff
@@ -233,8 +279,34 @@
          * @param value The value to to assign
          * @return This builder for chaining
          */
+        @Deprecated
         Builder assign(String role, boolean value);
 
+        /**
+         * Assigns this permission to the given role-template {@link Subject}.
+         *
+         * <p>If the given subject does not exist it will be created. Permission
+         * templates should not be assigned to regular subjects.</p>
+         *
+         * <p>It is recommended to use the standard role suggestions expressed
+         * as static parameters in {@link PermissionDescription}.</p>
+         *
+         * <p>Do not assign a permission to user, staff and admin at the same
+         * time but solve this with subject inheritance if possible.</p>
+         *
+         * <p><b>Note:</b> The permissions are only assigned during
+         * {@link #register()}.</p>
+         *
+         * @param role The role-template to assign the permission to. See
+         *             constants in {@link PermissionDescription} for common
+         *             roles (not exhaustive).
+         * @param value The value to to assign
+         * @return This builder for chaining
+         */
+        default Builder assign(String role, int value) {
+            return assign(role, value > 0);
```

Comment:
`assign(String, boolean)` is deprecated - why are you calling it?

Reply:
it's a fallback so that calling added api methods with a permissions plugin that hasn't been updated to specifically implement them won't cause breakage. These default implementations (and deprecated methods) would then be removed for API 8.